assume-role

```diff
@@ -159,12 +188,63 @@ assume-role(){
 
   if [ -z "$region" ]; then
     echo_out "region not defined"
-    return
+    return 1
   fi
 
   AWS_REGION="$region"
   AWS_DEFAULT_REGION="$region"
+}
 
+assume-role-with-saml() {
+  if [ -f "$SAML_FILE" ]; then
+    saml_user=$(sed -nE 's/username = (.+)/\1/gp' "$SAML_FILE")
+    saml_password=$(sed -nE 's/password = (.+)/\1/gp' "$SAML_FILE")
+  fi
+
+  echo_out "Gathering SAML credentials..."
+  if [ -n "$OUTPUT_TO_EVAL" ] && { [ -z "$saml_user" ] || [ -z "$saml_password" ] ;}; then
```

Comment:
@grahamjenson is this an ok if statement syntax? Not too familiar with doing more complex if statement logic in bash.

Reply:
if it works...

Reply:
tested locally, it works

Reply:
Agree that you wouldn't want a subshell in this context and so braces make sense, it's just more work to remember the gotchas with braces (i.e. you must have inner spaces, must ends with a semicolon).

Ultimately, nothing wrong with what you have here.